tests/php/test-amp-helper-functions.php

```diff
@@ -72,29 +94,103 @@ public function test_amp_get_slug() {
 	}
 
 	/**
-	 * Test amp_get_current_url().
+	 * Get data for testing amp_get_current_url().
 	 *
-	 * @covers ::amp_get_current_url()
+	 * @return array
 	 */
-	public function test_amp_get_current_url() {
-		$request_uris = [
-			'/foo',
-			'/bar?baz',
-			null,
-		];
+	public function get_amp_get_current_url_test_data() {
```

Comment:
This doesn't test any bogus values. I've found that `wp_parse_url()` can produce surprising results, so testing with empty strings, broken URLs, etc... seems necessary here.

Reply:
What kind of bogus return value for `home_url()` do you have in mind?

I suppose tests could be added for some IDN domains, Punycode, and IP addresses.

Reply:
Tests added in dd00644b20373dc374e034f460a3b2a7f32ecb2a.